Documentation/proposals/OneDotNetMSBuildProperties.md

```diff
@@ -0,0 +1,178 @@
+# One .NET: New MSBuild Properties
+
+One theme for Xamarin in .NET 6 is to simplify the cross-platform
+experience between iOS and Android. We would like to provide
+cross-platform alternatives to `AndroidManifest.xml` and `Info.plist`.
+The eventual goal would be to remove the need for these files for
+simple apps and project templates. Developers would only need to
+create them when a less widely used feature is needed. We would need
+to create item templates for `AndroidManifest.xml` and `Info.plist`
+when (and if) they are removed from project templates.
+
+As a starting point, we can add the following MSBuild properties to
+the Xamarin.Android and Xamarin.iOS SDKs:
+
+* `$(ApplicationId)` maps to `/manifest/@package` and
+  `CFBundleIdentifier`
+* `$(ApplicationVersion)` maps to `android:versionName` or
+  `CFBundleVersion`. This is a version string that must be incremented
+  for each iOS App Store or TestFlight submission.
+* `$(ApplicationVersionCode)` maps to `android:versionCode` (_Android
+  only)_. This is unfortunately an integer and must be incremented for
+  each Google Play submission. If omitted, the Android SDK can use
+  `$(ApplicationVersion)` for values up to 128.128.128.128. We can
+  pack most version strings into a 32-bit integer by default.
```

Comment:
Do we need to worry if users put a build number as the 4th number, where they could easily have an `$(ApplicationVersion)` like 1.0.1.1234?

Reply:
Personally I would not allow the reuse of `ApplicationVersion` for `ApplicationVersionCode`. Users should provide both, especially as google play has specific restrictions around these fields. We don't want to get into a situation where a user cannot upload an update because our version scheme changed (this has happened before). 

That said is this an opportunity to implement a feature where we pick up the `ApplicationVersion` from the assembly version...

Reply:
My initial thought was to require `$(ApplicationVersionCode)` to be set, and not try to pack versions like `1.0.0` into an `int`. @jonpryor was thinking we could make the properties identical for iOS/Android if we did it this way, but maybe we shouldn't do it?

> pick up the `ApplicationVersion` from the assembly version

There are 3 other properties that we might consider using:
* `$(Version)` used for NuGet package version
* `$(AssemblyVersion)` and `$(FileVersion)` into the generated `AssemblyInfo.cs`.

Maybe `$(ApplicationVersion)` should also set `$(FileVersion)` and `$(AssemblyVersion)` when they are blank? That probably makes sense.

I don't think we need to use `$(Version)` as you don't pack an app into a NuGet package. Maybe someone would try that, though?

Reply:
I think we should keep it separate and not pack. Remember the versionCode in the manifest also has some post processing via https://github.com/xamarin/xamarin-android/blob/master/Documentation/guides/building-apps/build-properties.md#androidversioncodepattern . 

The default pattern is `{abi}{versionCode:D5}` so we use the number in the manifest (e.g 1) and then combine that with other bits to produce a code based on the google recommendations https://github.com/xamarin/xamarin-android/blob/master/Documentation/guides/building-apps/~/android/deploy-test/building-apps/abi-specific-apks.md

Reply:
I changed this to be `$(AndroidVersionCode)`. We wouldn't try to pack the `$(ApplicationVersion)` into an integer.